python/lsst/faro/measurement/DetectorTableMeasurement.py

```diff
@@ -0,0 +1,129 @@
+import os
+
+import lsst.pipe.base as pipeBase
+from lsst.verify.tasks import MetricConnections
+import lsst.pex.config as pexConfig
+import lsst.geom
+from lsst.utils import getPackageDir
+
+from lsst.faro.base.CatalogMeasurementBase import CatalogMeasurementBaseTaskConfig, CatalogMeasurementBaseTask
+from lsst.pipe.tasks.loadReferenceCatalog import LoadReferenceCatalogConfig, LoadReferenceCatalogTask
+
+__all__ = ("DetectorTableMeasurementTaskConfig", "DetectorTableMeasurementTask")
+
+
+class DetectorTableMeasurementTaskConnections(MetricConnections,
+                                              dimensions=("instrument", "visit", "detector", "band"),
+                                              defaultTemplates={"refDataset": ""}):
+
+    catalog = pipeBase.connectionTypes.Input(doc="Source catalog for visit.",
+                                             dimensions=("instrument", "visit", "band"),
+                                             storageClass="DataFrame",
+                                             name="sourceTable_visit",
+                                             deferLoad=True)
+
+    refcat = pipeBase.connectionTypes.PrerequisiteInput(
+        doc="Reference catalog.",
+        name="{refDataset}",
+        storageClass="SimpleCatalog",
+        dimensions=("skypix",),
+        deferLoad=True,
+        multiple=True
+    )
+
+    measurement = pipeBase.connectionTypes.Output(doc="Per-detector measurement.",
+                                                  dimensions=("instrument", "visit", "detector", "band"),
+                                                  storageClass="MetricValue",
+                                                  name="metricvalue_{package}_{metric}")
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        if config.connections.refDataset == '':
+            self.prerequisiteInputs.remove("refcat")
+
+
+class DetectorTableMeasurementTaskConfig(CatalogMeasurementBaseTaskConfig,
+                                         pipelineConnections=DetectorTableMeasurementTaskConnections):
+    columns = pexConfig.Field(doc="Columns from sourceTable_visit to load.",
+                              dtype=str, default='coord_ra, coord_dec, detector')
+    refDataset = pexConfig.Field(doc="Reference dataset to use.",
+                                 dtype=str, default='')
+
+
+class DetectorTableMeasurementTask(CatalogMeasurementBaseTask):
+    ConfigClass = DetectorTableMeasurementTaskConfig
+    _DefaultName = "detectorTableMeasurementTask"
+
+    def run(self, catalog, dataIds, refcat, refcatCalib):
+        return self.measure.run(catalog, self.config.connections.metric, dataIds)
```

Comment:
I think it's so we can print log information as we process the quanta.

Reply:
It also doesn't pass refcatCalib.  I think it should probably pass both of those things on.

Reply:
Agree that we definitely want to pass `refcat` and `refcatCalib` in the `run` method. I think these should be keyword arguments because not all measurement tasks will need them.

Reply:
I _think_ the `dataIds` will not be needed. In other `faro` analysis contexts, the dataIds have been used to determine what bands are being analyzed, e.g., to compare with stellar locus (`WPerpTask` in `python/lsst/faro/measurement/TractMeasurementTasks.py`).

Reply:
If a task needs to determine the filter bands, you should pass in a `FilterLabel` or `list` of `FilterLabel`s.

Reply:
Thanks. That seems a cleaner approach. I've updated the following ticket to reflect this guidance.

https://jira.lsstcorp.org/browse/DM-31061